### DISCUSSION TITLE
In context responses.

### DISCUSSION THREAD

**Discussion by joshuachris2001** [imartinez/privateGPT#427] [Q&A]
Is it possible to access the previous responses, as context for future questions in a session, such as if I ask vicuna in `koboldcpp` such as. (the first two Q&A parts are from PrivateGPT for continuity)

> Question:

Tell me about the earth tides.

> Answer:

Tides are primarily caused by the gravitational pull of the moon and sun on the Earth's oceans, but other factors can also affect tides such as distance between the Earth, moon, and sun, shape and depth of the ocean floor, wind and weather patterns, Coriolis effect, and ocean currents. The gravitational pull of the moon and sun does not dictate when high or low tides occur. Instead, other forces such as the geography and shape of the Earth, and the depth of the world ocean control the tides on Earth.

> Question:

what other factors effect earths tides?

### Assistant:

According to the given context, there are several other factors that can also affect Earth's tides apart from the gravitational pull of moon and sun. These factors include distance between Earth, moon, and sun, shape and depth of ocean floor, wind and weather patterns, Coriolis effect, and ocean currents. Additionally, human activities such as dredging and construction can also affect tides in localized areas.

> Question:

as a list.

> Answer:

Sure! Here is a list of factors that can influence Earth's tides:

1. Gravitational pull of Moon and Sun
2. Distance between Earth, Moon, and Sun
3. Shape and Depth of Ocean Floor
4. Wind and Weather Patterns
5. Coriolis Effect
6. Ocean Currents
7. Dredging and Construction Activities
8. Other Natural Phenomena (e.g., El Niño)
9. Other Human Activities
[~~~~]

but if I ask the same model with the same above responses in privateGPT I get:
> Question:
as a list.

> Answer:
 Here is the list of contextual information you need to answer the question above:

’14
’15
’16
’17
’18
’19
’20
’21
’22
’23

**Comment by pseudotensor** [comment]
You  can do this with chatbot in https://github.com/h2oai/h2ogpt

**Reply by joshuachris2001** [reply]
I understand and have seen this repository instanced multiple times in PrivateGPT issue pages (mostly by you), I might try that at a later time but would like to focus on this repository for now. I prefer the cli interface over a web interface. and regretfully I only have cpu availability, I can't afford a decent gpu apart from a 2GB vram one.
